Sources/FusionLocation_Android/LocationManager+Android.swift

```diff
@@ -0,0 +1,144 @@
+import Java
+import Android
+import AndroidOS
+import AndroidApp
+import AndroidContent
+import AndroidLocation
+
+import FusionLocation_Common
+
+
+public class LocationManager {
+
+  typealias AndroidLocationManager = AndroidLocation.LocationManager
+
+  private var currentActivity: Activity? { Application.currentActivity }
+
+  private let locationManager: AndroidLocationManager?
+  private let locationListener: LocationListener
+  
+  public let usage: LocationUsage
+  
+  public required init(usage: LocationUsage) {
+    self.usage = usage
+
+    self.locationManager =
+      Application.currentActivity?.getSystemService(name: ContextStatic.LOCATION_SERVICE)
+      as? AndroidLocationManager
+  
+    self.locationListener = LocationListener()
+  }
+}
+
+extension LocationManager: LocationManagerProtocol {
+  public func requestAuthorization() {
+    currentActivity?.requestPermissions(      
+      permissions: [Manifest.permission.ACCESS_FINE_LOCATION], requestCode: 1111)
+  }
+
+  public func checkAuthorization() -> Bool {
+    guard
+      let status = currentActivity?.checkSelfPermission(
+        permission: Manifest.permission.ACCESS_FINE_LOCATION),
+
+      status == PackageManagerStatic.PERMISSION_GRANTED
+    else {
+      print("Permissions not granted. Requesting...")
+  
+      return false
+    }
+    print("Permissions granted")
+    return true
+  }
+
+  public func requestCurrentLocation(receiver: @escaping (FusionLocation_Common.Location?) -> Void) {    
+    guard
```

Comment:
The behaviour of this methods differs from the implementation on Apple. After requesting the current location, the Apple's version will receive a location asynchronously as soon as it's available, this version returns no location after a single call. Please, adjust the implementation.